lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -42,6 +43,85 @@ export interface GenericClusterProviderProps extends eks.ClusterOptions {
+     */
+    maxSize = new NumberConstraint(0, 100);
+
+    /**
+     * desiredSize has a maximum of 100 nodes per node group and can be as little as 0.
+     * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+     */
+    desiredSize = new NumberConstraint(0, 100);
+
+    /**
+     * amiReleaseVersion can be no less than 1 character long, and no greater than 63 characters long.
+     * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+     */
+    amiReleaseVersion = new StringConstraint(1, 63);
+}
+
+export class AutoscalingNodeGroupConstraints implements ConstraintsType<AutoscalingNodeGroup> {
+    /**
+    * id can be no less than 1 character long, and no greater than 63 characters long.
+    * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+    */
+    id = new StringConstraint(1, 63);
+
+    /**
+    * minSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    minSize = new NumberConstraint(0, 100);
+
+    /**
+    * maxSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    maxSize = new NumberConstraint(0, 100);
+
+    /**
+    * desiredSize has a maximum of 100 nodes per node group and can be as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster.
+    */
+    desiredSize = new NumberConstraint(0, 100);
+}
+
+export class FargateProfileConstraints implements ConstraintsType<FargateProfileOptions> {
+    /**
+     * fargateProfileNames can be no less than 1 character long, and no greater than 63 characters long.
+     * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+     */
+    fargateProfileName = new StringConstraint(1, 63);
+}
+
+export class GenericClusterPropsConstraints implements ConstraintsType<GenericClusterProviderProps> {
+    /**
+     * managedNodeGroups max size is 10 managed node groups per EKS cluster, and as little as 0.
+     * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster
+     */
+    managedNodeGroups = new ArrayConstraint(0, 10);
+    /**
+    * autoscalingNodeGroups max size is 10 managed node groups per EKS cluster, and as little as 0.
+    * https://aws.amazon.com/blogs/containers/eks-managed-node-groups/#:~:text=Additionally%2C%20there%20is%20a%20limit,on%20a%20given%20EKS%20cluster
```

Comment:
same comment: please use official AWS documentation and only apply hard limits (as opposed to soft limits). We can discuss.